### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits): the end of a functor

### PR DESCRIPTION
**Pull request by joelriou**
In this file, given a functor `F : Jᵒᵖ ⥤ J ⥤ C`, we define its end `end_ F`. It is a particular case of multiequalizer. The abbreviation `Wedge` for the corresponding category of cones is also introduced.

---

- [x] depends on: #17997

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dagurtomas** [comment]
Thanks!

maintainer merge

File: Mathlib/CategoryTheory/Limits/Shapes/End.lean [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  apply IsLimit.fac
+
+
+end IsLimit
+
+end Wedge
+
+section End
+
+/-- Given `F : Jᵒᵖ ⥤ J ⥤ C`, this property assets the existence of the end of `F`. -/
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Shapes/End.lean:104]:
```suggestion
/-- Given `F : Jᵒᵖ ⥤ J ⥤ C`, this property asserts the existence of the end of `F`. -/
```

File: Mathlib/CategoryTheory/Limits/Shapes/End.lean [hunk 2]
```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Limits.Shapes.Multiequalizer
+
+/-!
+# End and coends
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/End.lean:9]:
```suggestion
# Ends and coends
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by joelriou** [comment]
Thanks very much!

bors merge

**Pull request closed without merging**